tools/app/app/components/TreeMenu.tsx

```diff
@@ -12,38 +12,28 @@
 
 'use client';
 import React from 'react';
-import { Card, Project, WorkflowState } from '../lib/definitions';
 import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
-import { getStateColor } from '../lib/utils';
+import { getString, getStateColor } from '../lib/utils';
 import { Box, Stack, Typography } from '@mui/joy';
-import { sortItems } from '@cyberismocom/data-handler/utils/lexorank';
 import { Tree, NodeRendererProps, NodeApi } from 'react-arborist';
 import useResizeObserver from 'use-resize-observer';
 import { FiberManualRecord } from '@mui/icons-material';
-
-type CardNode = Card & {
-  workflowState: WorkflowState | undefined;
-};
+import {
+  ParseResult,
+  Result,
+} from '@cyberismocom/data-handler/utils/clingo-parser';
 
 type TreeMenuProps = {
   title: string;
-  project: Project;
   selectedCardKey: string | null;
-  onCardSelect?: (node: NodeApi<CardNode>) => void;
+  onCardSelect?: (node: NodeApi<Result>) => void;
   onMove?: (card: string, newParent: string, index: number) => void;
+  tree: ParseResult['results'];
 };
 
-// since we aren't using buckets, 1 means that missing ranks will be last
-function rankGetter(card: Card) {
-  return card.metadata?.rank || '1|z';
-}
-
-const RenderTree = (onCardSelect?: (node: NodeApi<CardNode>) => void) =>
-  function RenderNode({
-    node,
-    style,
-    dragHandle,
-  }: NodeRendererProps<CardNode>) {
+const RenderTree = (onCardSelect?: (node: NodeApi<Result>) => void) =>
+  function RenderNode({ node, style, dragHandle }: NodeRendererProps<Result>) {
+    const workflowStateCategory = getString(node.data, 'workflowStateCategory');
```

Comment:
Would it be cleaner if app backend (/api/tree/route.tsx) would parse the Result(s) into an object defined in definitions.ts? That way we would not need to access these named fields in app code and do null checks.

Reply:
I think that probably makes sense. Unfortunately these cannot use the definitions of data-handler but it also kind of makes sense. Then we'll have separate definitions for REST API / the actual data model. Though, I'd rather do these by just extending the parseResult. We'd also want to have some kind of validation that makes sure clingo actually returns what we expect it to return. Maybe this could be actually done at data-handler level. Do you think we gain anything if we rename the variables? E.g. I don't mind using "results" for the children, as long as we use it everywhere

Reply:
Some sort of validation would be good. All of this could also be on data-handler side, e.g. in addition to Calculate.runQuery we could have convenience functions like getTree(queryName) which would return a validated "Tree" object based on the given query.

As for naming, IMHO clearer is better; "result" is generic and will not give clues on what we expect to find there.

Reply:
I would be kind of against validating. I think we should define typescript types for them and the "runQuery"-method could the type, by assuming the data is correct. The reason for this is that these queries are like SQL queries, and we wouldn't validate SQL-queries because we assume they give us the right result, if the query is correct. I think same kind of thinking could be applied here. We could make tests for the queries though

Reply:
Yeah, above way of doing it sounds good!

Reply:
Did what I described so I resolved this